src/components/content/heroes/aurora/weather-easter-egg/daylight-percentage-calculator.ts

```diff
@@ -0,0 +1,41 @@
+export const getSunlightPercentage = (
+  sunriseTime: number,
+  sunsetTime: number,
+  time: number,
+) => {
+  const totalDaylightMinutes = sunsetTime - sunriseTime;
+  if (totalDaylightMinutes <= 0) {
+    return 20;
+  }
+  const currentMinutes = time - sunriseTime;
+  const sunlightPercentage = (currentMinutes / totalDaylightMinutes) * 100;
+
+  if (sunlightPercentage < 0) {
+    return 0;
+  }
+  if (sunlightPercentage > 100) {
+    return 100;
+  }
+  return Math.round(sunlightPercentage);
+};
+
+export const getNighttimePercentage = (
+  sunriseTime: number,
+  sunsetTime: number,
+  time: number,
+) => {
+  const totalNighttimeMinutes = sunriseTime - sunsetTime;
+  if (totalNighttimeMinutes <= 0) {
+    return 0;
```

Comment:
This does always return "0", right? sunriseTime as number is smaller than sunsetTime (I assume so, since the sun is shown correctly, and hence the daylightMinutes are greater than 0).